src/main/java/cms/gov/madie/measure/services/TestCaseLockService.java

```diff
@@ -113,4 +104,91 @@ public List<String> unlockByUser(String userName) {
     }
     return deleteMessages;
   }
+
+  public boolean lockTestCases(String measureId, List<String> testCaseIds, String userId) {
+    boolean failure = false;
+    List<TestCaseLock> locks = new ArrayList<>();
+    testCaseIds.forEach(
+        testCaseId -> {
+          validateMeasureAndTestCase(measureId, testCaseId);
+          TestCaseLock lock = makeNewLock(measureId, testCaseId, userId);
+          locks.add(lock);
+        });
+    List<TestCaseLock> insertedLocks = new ArrayList<>();
+    for (TestCaseLock lock : locks) {
```

Comment:
These two loops can be combined. Doing so would eliminate creation of `TestCaseLock` objects that are never used.